main.cpp

```diff
@@ -0,0 +1,451 @@
+            midiSendNoteOff(&midiMessageStruct);                        //Envío el mensaje de Note Off para no superponer notas 
+            midiSendNoteOn(&midiMessageStruct);                         //Envío el mensaje de Note On con el parámetro velocity proporcional a la intensidad del golpe
+            ledPad = LED_OFF;                                           //Apago el Led para indicar que se envió el mensaje correspondiente
+        }
+        else{}
+
+        if(buttonUpdate(&upButtonStruct) == BUTTON_PRESSED)             //Verifico si el pulsador upButton fué presionado
+        {
+            noteIndex++;                                                //Incremento el indice de navegación de notas
+            if (noteIndex >= numOfInstrumentNotes) noteIndex = 0;       //Controlo que el indice no se vaya de rango
+        }
+
+        if(buttonUpdate(&downButtonStruct) == BUTTON_PRESSED)           //Verifico si el pulsador downButton fué presionado
+        {
+            noteIndex--;                                                //Decremento el indice de navegación de notas
+            if (noteIndex < 0) noteIndex = numOfInstrumentNotes - 1;    //Controlo que el indice no se vaya de rango
+        }
+
+    }
+
+}
+
+/*********************************************************************
+ *=====[Implementación de funciones públicas]=========================
+ ********************************************************************/
+
+uint8_t buttonUpdate(button_t* button)
+{
+    button->currentState = button->alias->read();           //Leo el estado actual del pulsador
+    if (button->currentState != button->lastState)          //Verifico si el estado ha cambiado
+    {
+        wait_us(DEBOUNCE_DELAY_MS * 1000);                  //Espero un tiempo prudente de rebote
+        if (button->currentState == button->alias->read())  //Confirmo si realmente hubo un cambio de estado
+        {
+                button->lastState = button->currentState;   //Actualizo el ultimo estado relevado
+                return  button->currentState;               //Devuelvo el estado relevado          
+        }      
+    }
+    return BUTTON_BOUNCING;                                 //Devuelvo el estado de rebote
+
+}
+
+void outputsInit()
+{
+    ledPad = LED_OFF;   //Inicializo el led del drum pad apagado
+}
+
+void calculateSlopeIntercept()
+{
+    slope = (float)DELTA_VEL / DELTA_VOLT;                  /**< Pendiente de la recta de conversión */
+    intercept = MIN_VEL - PIEZO_THRESHOLD_mV * slope;       /**< Ordenada al origen de la recta de conversión */ 
+}
+
+uint8_t piezoUpdate(piezo_t* piezo)
+{
+    float piezoRead = 0.0;                                              /**< Valor leido del transductor piezoeléctrico  */
+    float piezoMax = 0.0;                                               /**< Máximo valor leido del transductor piezoeléctrico */
```

Comment:
Por qué floating point si tenés que obtener y entregar un entero ?
Tenés idea de todo lo que pasa debajo tuyo para resolver eso ? Zafás si el micro tiene FPU y linkeás con hard floating-point; así y todo la aritmética entera es muchísimo más rápida.

Reply:
Honestamente no tengo mucha idea, por no decir "no tengo la más mínima idea"...que lo estoy diciendo.... Si tenés alguna fuente de información para desasnarme será bienvenida!

Reply:
Jack Ganssle, un servidor, Google is your friend. Si Google no te lleva a esos 2 destinos, te lo paso.